src/cli_callback.c

```diff
@@ -555,129 +485,49 @@ static int download_file(storj_env_t *env, char *bucket_id,
-
-    cli_api_t *cli_api = req->handle;
-    cli_api->last_cmd_req = cli_api->curr_cmd_req;
-    cli_api->rcvd_cmd_resp = "list-mirrors-resp";
-
-    if (req->status_code != 200) {
-        printf("Request failed with status code: %i\n",
-               req->status_code);
-        goto cleanup;
-    }
-
-    if (req->response == NULL) {
-        free(req);
-        free(work_req);
-        printf("Failed to list mirrors.\n");
-        goto cleanup;
-    }
-
-    int num_mirrors = json_object_array_length(req->response);
-
-    struct json_object *shard;
-    struct json_object *established;
-    struct json_object *available;
-    struct json_object *item;
-    struct json_object *hash;
-    struct json_object *contract;
-    struct json_object *address;
-    struct json_object *port;
-    struct json_object *node_id;
-
-    for (int i = 0; i < num_mirrors; i++) {
-        shard = json_object_array_get_idx(req->response, i);
-        json_object_object_get_ex(shard, "established",
-                                 &established);
-        int num_established =
-            json_object_array_length(established);
-        for (int j = 0; j < num_established; j++) {
-            item = json_object_array_get_idx(established, j);
-            if (j == 0) {
-                json_object_object_get_ex(item, "shardHash",
-                                          &hash);
-                printf("Shard %i: %s\n", i, json_object_get_string(hash));
-            }
-            json_object_object_get_ex(item, "contract", &contract);
-            json_object_object_get_ex(contract, "farmer_id", &node_id);
-
-            const char *node_id_str = json_object_get_string(node_id);
-            printf("\tnodeID: %s\n", node_id_str);
-        }
-        printf("\n\n");
-    }
-
-    json_object_put(req->response);
-
-    queue_next_cmd_req(cli_api);
-cleanup:
-    free(req->path);
-    free(req);
-    free(work_req);
-}
-
 static void delete_file_callback(uv_work_t *work_req, int status)
 {
     assert(status == 0);
-    json_request_t *req = work_req->data;
+    delete_file_request_t *req = work_req->data;
 
     cli_api_t *cli_api = req->handle;
     cli_api->last_cmd_req = cli_api->curr_cmd_req;
     cli_api->rcvd_cmd_resp = "remove-file-resp";
 
-    if (req->status_code == 200 || req->status_code == 204) {
-        printf("File was successfully removed from bucket.\n");
-    } else if (req->status_code == 401) {
-        printf("Invalid user credentials.\n");
-        goto cleanup;
-    } else if (req->status_code == 403) {
-        printf("Forbidden, user not active.\n");
-        goto cleanup;
-    } else {
-        printf("Failed to remove file from bucket. (%i)\n", req->status_code);
+    if (req->error_code) {
+        printf("Failed to remove file from bucket: (%s)\n",
+               storj_strerror(req->error_code));
         goto cleanup;
     }
 
     json_object_put(req->response);
 
     queue_next_cmd_req(cli_api);
 cleanup:
-    free(req->path);
```

Comment:
why have a separate `storj_free_delete_file_request` function instead of freeing here under `cleanup:` like with most of the functions?

Reply:
I guess there is `storj_free_get_buckets_request`, so it's understandable

Reply:
:+1:

Reply:
There were a few free functions before this work but not every request type had a free function.